### PR TITLE
Pull req for project authorization / Kat and Liza

### PR DESCRIPTION
**Pull request by Katsiez**
Link to my deployed project:
Backend = https://project-auth-liza-kat.herokuapp.com/
Frontend = https://practical-mestorf-d4dfd7.netlify.app/sessions

My reflections on how this week's project turned out:
Turned out good after a lot of hours of working on it together. Much more complicated than we anticipated! :D

Things I'd like to have clarified or explained in more detail:
How I still have a functioning brain after all this mumbo jumbo lol.

### CODE REVIEW

File: backend/server.js [hunk 1]
```diff
@@ -1,29 +1,166 @@
...
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+import crypto from "crypto";
+import bcrypt from "bcrypt-nodejs";
+import endpoints from "express-list-endpoints";
```
Comment (jonnaru) [inline backend/server.js:7]: Cool, didn't know about this 😎

File: backend/server.js [hunk 2]
```diff
@@ -1,29 +1,166 @@
...
+      password: bcrypt.hashSync(password)
+    })
+    await user.save()
+    res.status(201).json({
+      message: 'User created!',
+      id: user._id,
+      accessToken: user.accessToken,
+      name: user.name,
+      email: user.email,
+      password: user.password
```
Comment (jonnaru) [inline backend/server.js:94]: Are you sure sending back password is a good idea, even if it's hashed? 🧐

File: frontend/src/App.js [hunk 3]
```diff
@@ -1,9 +1,34 @@
...
+            <SignUp />
+          </Route>
+          <Route path="/sessions" exact>
+            <Login />
+          </Route>
+        </Switch>
+      </BrowserRouter> 
+    </Provider>
+  );
+};
```
Comment (jonnaru) [inline frontend/src/App.js:34]: Redux and router!!! 🤩

File: frontend/src/components/Home.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        >
+          <Text>
+            <TextAnimation>
+              <span>
+                Welcome to XYZ. Let's get you logged in! Don't have an account
+                yet? Sign up and get started.
+              </span>
+            </TextAnimation>
+          </Text>
+        </AnimatedBg>
```
Comment (jonnaru) [inline frontend/src/components/Home.js:28]: Pretty ⭐️

File: frontend/src/components/Login.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        );
+        setLogInSuccess(true);
+      })
+      .catch(() => {
+        setLogInFailed(true);
+      })
+      .finally(() => {
+        setEmail("");
+        setPassword("");
+      });
```
Comment (jonnaru) [inline frontend/src/components/Login.js:50]: Good idea!

File: frontend/src/components/UserProfile.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  );
+};
+
+const Image = styled.main`
+  background-image: url("${process.env.PUBLIC_URL + "/flower.jpg"}");
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  background-size: cover;
+`;
```
Comment (jonnaru) [inline frontend/src/components/UserProfile.js:75]: If you are using the same styling in more components than one, like this one, maybe you can move the styling out from the component and then import this image styling 😍

**Review by jonnaru — APPROVED** [review]
Hey! Great work 🤩 Really inspiring. Loved reading your code. Super cute login-page. I think there is a problem, perhaps between router and netlify. Sometimes I get this error-messages from netlify 🤷‍♀️ But I got all the way to the secret message and loged out 🙌

File: frontend/src/components/Home.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        </Redirect>
+        <Redirect to="/users">
+          <SubmitButton title="Sign Up" />
+        </Redirect>
+      </ButtonContainer>
+    </Image>
+  );
+};
+
+const Image = styled.main`
```
Comment (KarolCislowski) [inline frontend/src/components/Home.js:43]: nice that you used styled-components ❤️

File: frontend/src/reducers/user.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    logout: (state, action) => {
+      console.log("Logging out");
+      state.login.userId = 0;
+      state.login.email = "";
+      state.login.name = "";
+      state.login.accessToken = null;
+      state.login.secretMessage = "";
+    },
+  },
+});
```
Comment (KarolCislowski) [inline frontend/src/reducers/user.js:49]: Reducer is a great place to manage your users, but you need to store some data in the local storage to keep your users logged in after they refresh the page. 🤓

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
It is a very nice project 🦄 

Please fix the losing user on refresh issue and it will be a magical app 🌈

**Review by KarolCislowski — APPROVED** [review]
After the discussion with teachers about this assignment, we decided to approve this project without implementing the local storage functionality. 

However for the future, please remember to store your users' authentication data in your browser to make your UX good.

Good luck with your Final Project! 🍀

**Pull request closed without merging**